### DISCUSSION TITLE
How to get transaction status in Cardano?

### DISCUSSION THREAD

**Discussion by jasoncsli** [IntersectMBO/cardano-node#5255] [Q&A] [answered]
Are there any failed transactions in the block for Cardano? I am little bit of confused because I can't really find the status for the transaction.

**Comment by mkoura** [comment, marked as answer]
You can check it in Cardano Blockchain Explorer by transaction id, eg. https://explorer.cardano.org/en/transaction?id=470b40527b166716a57b9e931a968b82fb3b45682b5332f2c4321946004daf81

If you are using `cardano-cli`, you can submit a transaction and check following
* if you don't see any error reported during submit, the tx was likely submitted successfully
* you can query if the tx is in mempool using `cardano-cli query tx-mempool exists <TxId>`
* you can query that the tx was removed from mempool and included in a block by querying tx inputs using `cardano-cli query utxo --tx-in`
  * if the tx is no longer in mempool and the tx inputs cannot be queried (were already spent), the transaction was successfully included in a block
  * if the tx is no longer in mempool and the tx inputs can be queried, the tx was either not submitted successfully or got reverted for some reason (short lived fork etc.). You can re-submit the tx.

**Comment by UbbeWolf** [comment]
Increases knowledge. Thank you
